### PR TITLE
build: properly lint tooling source code

### PR DESCRIPTION
**Pull request by DevVersion**
* Fixes a glob that is responsible for IDE's TypeScript service and also for TSLint
* Fixes strict null check errors in the screenshot app. Currently the app can't be compiled properly since strict null checks had been introduced there as well.

**Note**: Two remaining lint issues coming from https://github.com/angular/material2/pull/6044

### CODE REVIEW

File: tools/screenshot-test/src/app/firebase.service.ts [hunk 1]
```diff
@@ -71,10 +71,7 @@ export class FirebaseService {
             this._readResults(childSnapshot);
             break;
         }
-        counter++;
-        if (counter === snapshot.numChildren()) {
-          return true;
-        }
+        return ++counter === snapshot.numChildren();
```
Comment (jelbourn) [inline tools/screenshot-test/src/app/firebase.service.ts]: I prefer not to use increment/decrement expressions as part of a larger expression where the order of the increment/decrement would matter
(here and elsewhere)

Comment (DevVersion) [inline tools/screenshot-test/src/app/firebase.service.ts]: Not sure. I thought it's pretty basic a pretty JavaScript syntax. I'm fine changing it if you prefer it though.

Comment (DevVersion) [inline tools/screenshot-test/src/app/firebase.service.ts]: Changed it. Please have another look.

Comment (jelbourn) [inline tools/screenshot-test/src/app/firebase.service.ts]: It's basic syntax in Java, C++, C#... and in all of those languages, I see this as potentially confusing.

**Pull request merged**